### DISCUSSION TITLE
Support for angular

### DISCUSSION THREAD

**Discussion by srvraj311** [shadcn-ui/ui#2625] [Ideas]
Is there any existing or future plan for support for angular.

**Comment by XhstormR** [comment]
+1

**Comment by gregermendle** [comment]
The underlying headless components are Radix and are written in React. I believe your best option is to use https://www.spartan.ng/components which looks very similar.

**Reply by msaeedsaeedi** [reply]
The Spartan project is still in development and is missing quite a few key components. The installation process can also be a bit tricky for beginners. On top of that, I haven’t heard back from them on Discord after reaching out four days ago, which makes it feel a bit challenging to get support or collaboration.

**Reply by guibzo** [reply]
spartan seems a bit abandoned, unfortunately

**Comment by yagobski** [comment]
+1

**Comment by im-ashar** [comment]
+1

**Comment by chrehman** [comment]
+1

**Comment by Zac10ck** [comment]
+ 1

**Comment by beamerboi** [comment]
+1

**Comment by ZakeriaHirsi** [comment]
+1

**Comment by IvanMamaniPerez** [comment]
+1

**Comment by msaeedsaeedi** [comment]
+1

**Comment by minhperry** [comment]
+1

**Comment by ShadyManu** [comment]
+1

**Comment by deivisonrpg** [comment]
+1

**Comment by ka1ea6** [comment]
+1

**Comment by deivisonrpg** [comment]
https://www.originui-ng.com/

**Comment by bogdanliusik** [comment]
+1

**Comment by miltostave** [comment]
+1

**Comment by Lerado** [comment]
+1

**Comment by Khumozin** [comment]
+1

**Comment by dvzrd** [comment]
I was cooking an idea of separating the styles into a css and cva design system to decouple the ui, in the hopes of applying the same styles to a separate ui using angular and react (as a POC for a framework agnostic solution). Based on that work, I realized it would be a high effort undertaking that prevented me from working on actual product ideas, so I left it on the back burner for now. 

Then there was also the problem of having the tailwind dependency, which if you want a true agnostic solution you would essentially need to have stylesheets that are separate from tailwind but can also be imported inside tailwind layers (for those that wanted a tailwind solution). I had most of this setup and working with the latest versions, but the code needs to be cleaned up before it can actually be used by others.

I paused the idea for now because I couldn't afford to get distracted and shadcn/ui and tailwind works for my needs right now, but I wouldn't mind coming back around to it. If anyone's interested, I can open source a POC with 1 or 2 components using angular and react, but it would need to be baked more and optimized with build scripts and tree shaking for the decoupled css before it can be used in production.

**Comment by johngrimsey** [comment]
+111111

**Comment by Picoquio** [comment]
I will never understand the first class treatment that React has been given by the majority of the ecosystem while superior tools like angular get forgotten

**Comment by mgShehata** [comment]
@Picoquio its sad but i guess it supply/demand kind of things
lets hope we get it soon since there is angular edition of radix now

**Comment by guibzo** [comment]
+1

**Comment by babakamyljanovssw** [comment]
+1

**Comment by 0zminDev** [comment]
+1

**Comment by ankan-withadream** [comment]
+1

**Comment by devkazuto** [comment]
+1

**Comment by jg4n** [comment]
+1

**Comment by eremeevfd** [comment]
+1

**Comment by Luizgomess** [comment]
https://github.com/zard-ui/zardui, PROD: https://zardui.com/, DEV: https://dev.zardui.com/ , Hello @srvraj311  I'm one of the founders of zard-ui, where we aim to have an alternative to shadcn-ui for Angular. Currently, many details are under development, but we're moving very quickly. Our focus is to combine simplicity, a good dev experience, and the shadcn UI.

**Reply by Khumozin** [reply]
There's spartan ui, and it's almost production ready.

https://spartan.ng/

**Reply by Luizgomess** [reply]
@Khumozin I didn't like Spartan, I think it's too much code to do simple things, I didn't find several things practical, that's why I created Zard/ui, considering how poor the Angular ecosystem is for component libs, if a third party comes along trying to do the same thing, the community will be grateful.

**Reply by Luizgomess** [reply]
### An example to illustrate why I didn't really like Spartan and wanted to start my own lib.

 <b>For do this</b>
<img width="827" height="311" alt="image" src="https://github.com/user-attachments/assets/969f77cd-364b-4a39-94bc-0299b3ebd816" />
<br>
In Spartan you add this:
<br>
<img width="855" height="666" alt="image" src="https://github.com/user-attachments/assets/5b42fbf7-f90b-4d7b-aea4-c0c47647a80b" />
 <br>
and this too
<br>
<img width="771" height="853" alt="image" src="https://github.com/user-attachments/assets/20204960-5867-4ce8-b984-dfa45c53f633" />
<br>
At Zard/ui our approach is to have THE SAME RESULT, with just this...
<br>
<img width="1177" height="763" alt="image" src="https://github.com/user-attachments/assets/ea9b0273-c1cf-4e0e-861d-ea969995e0c6" />

We're still in alpha, but we're moving forward very quickly, as we haven't even been on the project for a year. Much of the component design was designed around the real pain points of several Angular developers who came together to make this project successful. For me, Spartan has a great design, but the development experience isn't great.

and being very honest the angular ecosystem is already so involved with angular material, primeng, ng-zorro and some others that have their limitations in terms of customization, the more well-structured libs in this style exist, the community can only be grateful, because in this matter React is far ahead.

**Reply by myarovenko-mspt** [reply]
well, to manage imports you could've combined them into modules yourself and just import one module, isn't that the whole point of owning components code?

**Reply by Luizgomess** [reply]
@myarovenko-mspt For some zard components, we give the option to import the module with all the components as well, if you don't want to import them 1 by 1
<img width="1430" height="860" alt="image" src="https://github.com/user-attachments/assets/1572e480-c1c6-4c4d-8688-1854e71026bf" />

**Reply by tienthanhjlw** [reply]
many thanks to your team!
